lib/src/parallax.dart

```diff
@@ -235,20 +240,33 @@ enum LayerFill { height, width, none }
 /// layer moves with different velocities to give an effect of depth.
 class Parallax {
   Vector2 baseVelocity;
+  Vector2 _size = Vector2.zero();
+  Rect _clipRect;
   final List<ParallaxLayer> layers;
 
   Parallax(
-    this.layers, {
+    this.layers,
+    Vector2 size, {
     Vector2 baseVelocity,
-  }) : baseVelocity = baseVelocity ?? Vector2.zero();
+  })  : assert(layers != null),
+        assert(size != null) {
+    this.baseVelocity = baseVelocity ?? Vector2.zero();
+    resize(size);
+  }
 
   /// The base offset of the parallax, can be used in an outer update loop
   /// if you want to transition the parallax to a certain position.
   Vector2 currentOffset() => layers[0].currentOffset();
 
   /// If the `ParallaxComponent` isn't used your own wrapper needs to call this
   /// on creation.
-  void resize(Vector2 size) => layers.forEach((layer) => layer.resize(size));
+  void resize(Vector2 newSize) {
+    if (newSize.x != _size.x || newSize.y != _size.y) {
```

Comment:
dont we have equality operator for vector2?

Reply:
Nope

Reply:
I think we should add it

Reply:
https://api.dart.dev/stable/2.10.4/dart-core/Object/operator_equals.html

Reply:
Apparently we can't override this kind of operator in extensions :(

![Captura de Tela 2021-02-02 às 20 56 32](https://user-images.githubusercontent.com/835641/106678365-54150f00-6599-11eb-95da-1e539e59e6b9.png)

Reply:
ah so sad!

Reply:
but that is because they already have it https://api.flutter.dev/flutter/vector_math/Vector2/operator_equals.html

Reply:
Done